bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCrockpotHandler.java

```diff
@@ -56,44 +55,44 @@
 
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_CROCKPOT);
 
-    private final Map<String, Boolean> subscriptionState = new HashMap<>();
-    private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
+    private final UpnpIOService service;
+
+    private final Object upnpLock = new Object();
+    private final Object jobLock = new Object();
 
-    private UpnpIOService service;
+    private final Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<>());
 
     private WemoHttpCall wemoCall;
 
-    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable String host;
 
-    private final Runnable refreshRunnable = () -> {
-        updateWemoState();
-        if (!isUpnpDeviceRegistered()) {
-            logger.debug("WeMo UPnP device {} not yet registered", getUDN());
-        } else {
-            onSubscription();
-        }
-    };
+    private Map<String, Boolean> subscriptionState = new HashMap<>();
+
+    private @Nullable ScheduledFuture<?> pollingJob;
 
     public WemoCrockpotHandler(Thing thing, UpnpIOService upnpIOService, WemoHttpCall wemoHttpCaller) {
         super(thing, wemoHttpCaller);
 
         this.wemoCall = wemoHttpCaller;
         this.service = upnpIOService;
 
-        logger.debug("Creating a WemoCrockpotHandler for thing '{}'", getThing().getUID());
+        logger.debug("Creating a WemoCrockpotHandler for thing '{}' with IP '{}'", getThing().getUID(), host);
```

Comment:
Seems like left-over as host will no longer be initialized at this point?

Reply:
Removed in all handler classes.